### PR TITLE
reenable the controller default relist interval

### LINKED ISSUE #2119: Change default relist and add a config flag
I might not be looking at the right spot, but from looking at the code it appears that we will automatically re-list the broker's catalog every 15 minutes and there is no way to set this default timeout to something else (unless you do it on a per-broker basis as you register them).

While I know there are some brokers that change their catalogs very often (and in those cases I suspect they'll set this re-list timeout to a much lower value for their brokers), it seems like 15 minutes is actually really low/quick - in our experience brokers don't change their catalogs that often.

So, there are two asks:
1 - can we change the default from 15 minutes to something longer, like a few hours?
2 - regardless of number 1, can we get a config option so we can set if for all brokers as they are registered rather than having to set it on each broker's yaml?

ping @MHBauer

### PR DESCRIPTION
**Pull request by MHBauer**
- dump apiserver defaults + validation
 - dump checks in controller for a set value

resolves #2119

/cc @duglin 
/assign @duglin

### CODE REVIEW

File: pkg/apis/servicecatalog/validation/broker.go [hunk 1]
```diff
@@ -188,20 +188,6 @@ func validateCommonServiceBrokerSpec(spec *sc.CommonServiceBrokerSpec, fldPath *
...
 	}
 
-	if spec.RelistBehavior == sc.ServiceBrokerRelistBehaviorDuration && spec.RelistDuration == nil {
-		commonErrs = append(
-			commonErrs,
-			field.Required(fldPath.Child("relistDuration"), "relistDuration must be set if relistBehavior is set to Duration"),
-		)
-	}
-
-	if spec.RelistBehavior == sc.ServiceBrokerRelistBehaviorManual && spec.RelistDuration != nil {
```
Comment (carolynvs) [inline pkg/apis/servicecatalog/validation/broker.go]: I don't understand why this check for manual was removed?

Comment (carolynvs) [inline pkg/apis/servicecatalog/validation/broker.go]: What's the defined behavior for a manual relist? I don't quite understand why your change makes it okay to say manual and set a duration.

Comment (MHBauer) [reply pkg/apis/servicecatalog/validation/broker.go]: Manual is "only respond to an increase in the relist-requests, no automated pinging of the broker" 

My opinion is that this causes unecessary friction if switching from duration to manual, and that while it doesn't make much sense to explicitly specify manual and duration, it shouldn't be a hard error. Manual should override everything else.

Comment (carolynvs) [reply pkg/apis/servicecatalog/validation/broker.go]: Ah, that makes sense thanks! 👍

**Comment by duglin** [comment]
LGTM

**Pull request merged**